### ISSUE TITLE
Split up MainPass3dNode

### ISSUE THREAD

**Issue by JMS55** [bevyengine/bevy#8089]
## What problem does this solve or what need does it fill?

Upcoming render features want to optionally insert rendering logic in between the opaque/alpha/transparent phases.

## What solution would you like?

Split MainPass3dNode into separate nodes.

## Additional context

* https://github.com/bevyengine/bevy/issues/8008
* https://github.com/bevyengine/bevy/pull/8015

**Comment by superdump** [comment]
Yup. Though importantly they need to reuse render passes where possible. Opaque and alpha mask should reuse the same render pass for example.

**Comment by JMS55** [comment]
We don't currently do that, but as part of splitting the nodes up we could do so.
